### PR TITLE
[Connector API] Add update filtering validation and activate draft endpoints

### PR DESCRIPTION
**Pull request by jedrazb**
### Changes

- Add 2 new endpoints to handle updating draft validation (service endpoint) and activating the filtering draft (management endpoint)
- Added checks to make sure the only draft in `valid` state can be activated
- Unit tests, yaml tests, bwc tests added 

Endpoints
```
PUT _connector/{id}/_filtering/_validation
{
   "state": ..., "errors": [...]
}

PUT _connector/{id}/_filtering/_activate
```

### CODE REVIEW

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java [hunk 1]
```diff
@@ -644,6 +645,111 @@ public void updateConnectorFilteringDraft(
         }
     }
 
+    /**
+     * Updates the {@link FilteringValidationInfo} of the draft {@link ConnectorFiltering} property of a {@link Connector}.
+     *
+     * @param connectorId  Request for updating connector filtering property.
```
Comment (timgrein) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java:651]:
```suggestion
     * @param connectorId  Request for updating {@link ConnectorFiltering}.
```

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java [hunk 2]
```diff
@@ -644,6 +645,111 @@ public void updateConnectorFilteringDraft(
...
+                l.onFailure(e);
+            }
+        }));
+
+    }
+
+    /**
+     * Activates the draft {@link ConnectorFiltering} property of a {@link Connector}.
+     *
+     * @param connectorId  Request for updating connector filtering property.
```
Comment (timgrein) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java:699]:
```suggestion
     * @param connectorId  Request for updating {@link ConnectorFiltering} property.
```

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java [hunk 3]
```diff
@@ -644,6 +645,111 @@ public void updateConnectorFilteringDraft(
...
+                // Connectors represent their filtering configuration as a singleton list
+                ConnectorFiltering connectorFilteringSingleton = connectorFilteringList.get(0);
+
+                FilteringValidationState currentValidationState = connectorFilteringSingleton.getDraft()
+                    .getFilteringValidationInfo()
+                    .getValidationState();
+
+                if (currentValidationState != FilteringValidationState.VALID) {
+                    throw new ElasticsearchStatusException(
+                        "Filtering draft needs to be validated by the framework before activation. "
```
Comment (timgrein) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java:718]:
```suggestion
                        "Filtering draft needs to be validated by the connector service before activation. "
```

I think being explicit with "connector service" or "connector framework" is a bit better.

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/filtering/FilteringValidation.java [hunk 4]
```diff
@@ -69,6 +69,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.stringListField(IDS_FIELD.getPreferredName(), ids);
             builder.stringListField(MESSAGES_FIELD.getPreferredName(), messages);
         }
+        builder.endObject();
```
Comment (timgrein) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/filtering/FilteringValidation.java:72]: Did we've a bug here? If yes, does this need to be backported?

**Review by timgrein — APPROVED** [review]
LGTM, some nits and one question, if we need to backport something

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/filtering/FilteringValidation.java (hunk 4 above)
Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/filtering/FilteringValidation.java:72]: yes we did, yeah I'm looking into creating a backport PR !

Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/filtering/FilteringValidation.java:72]: https://github.com/elastic/elasticsearch/pull/107467

**Pull request merged**